### DISCUSSION TITLE
ord_get() extract Eigenvalues for all the samples

### DISCUSSION THREAD

**Discussion by Leran10** [david-barnett/microViz#55] [Q&A] [answered]
Hi,

I'm trying to extract eigenvalues for every sample with ord_get() function, so that I can use PC1 values of each sample to build a lm(diagnosis ~ PC1) model to get a p value for the PCA plot. But when I run ord_get() on ps_extra object, I only get:

```
Call: rda(formula = OTU ~ 1, data = data)

              Inertia Rank
Total           309.8     
Unconstrained   309.8  116
Inertia is variance 

Eigenvalues for unconstrained axes:
  PC1   PC2   PC3   PC4   PC5   PC6   PC7   PC8 
50.56 26.64 20.40 14.73 11.55 10.95  9.00  8.43 
**(Showing 8 of 116 unconstrained eigenvalues)**
```

So I wondered how can I get PC values of other 115 samples? Or do you have any suggestion on how to get p value for PCA plot?

Thanks!
Leran

**Comment by david-barnett** [comment, marked as answer]
Hi Leran

For a p-value for an ordination plot (in the sense of e.g. assessing the significance of an overall difference in microbiome composition between two groups visualized on a PCA plot), you might consider PERMANOVA, using `dist_permanova`, or the underlying vegan function `adonis2`.

For the specific question about getting "PC values" for each sample, you can look at the site scores using the vegan package function `scores` (on the ord_get output). Beware that this is definitely not the same thing as eigenvalues, which is shown in the printout of the ord_get output.

I suspect you would want PERMANOVA, unless you have a particular reason to only be interested in PC1 here.
